keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/DeliveryArtifact.kt

```diff
@@ -32,6 +41,8 @@ abstract class DeliveryArtifact {
   abstract val reference: String // friendly reference to use within a delivery config
   abstract val deliveryConfigName: String? // the delivery config this artifact is a part of
   open val statuses: Set<ArtifactStatus> = emptySet()
+  open val branch: String? = null
+  open val sortBy: ArtifactSortByMethod = VERSION
 
```

Comment:
In order to validate when branch-and-timestamp is specified, you can do something like: https://github.com/spinnaker/keel/blob/master/keel-ec2-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt#L201-L211

it's super easy

Reply:
Nice! Will try that.

Reply:
@emjburns I tried that, but ran into issues with class initialization order: https://kotlinlang.org/docs/reference/classes.html#derived-class-initialization-order

I spent some time trying to get it to work, but it felt brittle and so I'd rather add a validator separately in a follow-up PR. Hope that's OK.

Reply:
eww yuck. that's fine

Reply:
The good news is we don't need any validation anymore as I'm now auto-detecting the strategy. 🙂